docs/contributing-design.rst

```diff
@@ -0,0 +1,105 @@
+The Design of dd-trace-py
+=========================
+
+Parts of the Library
+--------------------
+
+When designing a change, one of the first decisions to make is where it should be made. This is an overview
+of the main functional areas of the library.
+
+A **product** is a unit of code within the library that implements functionality specific to a small set of
+customer-facing Datadog products. Examples include the `appsec module <https://github.com/DataDog/dd-trace-py/tree/1.x/ddtrace/appsec>`_
+implementing functionality for `Application Security Management <https://www.datadoghq.com/product/application-security-management/>`_
```

Comment:
This redirects to https://www.datadoghq.com/product/app-and-api-protection/ as the product is now named `App and API Protection`